### ISSUE TITLE
Auto complete not works in Rust

### ISSUE THREAD

**Issue by siygle** [zed-industries/zed#4684]
### Check for existing issues

- [X] Completed

### Describe the bug / provide steps to reproduce it

Auto-complete seems not work in Rust (but it's ok when I switch to TS) and the rust-analyzer version under `~/Library/Application Support/Zed/languages/rust-analyze` is `rust-analyzer-2023-06-12`.

*BTW, thanks for this awesome project, can't wait to use it on my daily work!* 🙌 
*Love it already support Vim mode already. 😄 *

### Environment

Zed: v0.90.2 (stable)
OS: macOS 13.4.0
Memory: 16 GiB
Architecture: aarch64

### If applicable, add mockups / screenshots to help explain present your vision of the feature

![SCR-20230618-ef6](https://github.com/zed-industries/community/assets/173408/36f37543-a878-4e2b-b09b-e7782f900ee2)
*Auto-complete did not show up*

### If applicable, attach your `~/Library/Logs/Zed/Zed.log` file to this issue.
If you only need the most recent lines, you can run the `zed: open log` command palette action to see the last 1000.

[Zed.log](https://github.com/zed-industries/community/files/11780429/Zed.log)

**Comment by Orange4Broom** [comment]
Hi, I have encountered the same issue. However, I am using TypeScript, and the solutions provided here are not working for me. I have also configured settings. I'm running Zed 0.120.6 on macOS 14.2.1 (23C71).

**Comment by dailydaniel** [comment]
+1 zed 0.125.3 macos 14.3 m1, it's ok in python and elixir, but doesn't work in rust.

**Comment by zlElo** [comment]
Hey, hey,
I disabled GitHub copilot, reopened zed and now everything is working. For me, GitHub copilot is not necessary, so it's no problem to disable it. 
Greets

**Comment by babylard** [comment]
> +1 zed 0.125.3 macos 14.3 m1, it's ok in python and elixir, but doesn't work in rust.

Yeah i'm having the same issue, all the GitHub issues I've check don't seem to help either. I'll just deal with it for now I guess. Still an awesome editor.

**Comment by ThatOnePhysicist** [comment]
I think I found something that works. I have my folders nested like so, 

```
rust/
├── bookprojects
│   ├── Chapter2
│   │   └── guessing_game
│   └── Chapter5
│       └── Structs
├── project1
│   └── project1_name
│       ├── Cargo.lock
│       ├── Cargo.toml
│       ├── src/
│       └── target/
├── project2
│   └── project2_name
│       ├── Cargo.lock
│       ├── Cargo.toml
│       ├── src/
│       ├── target/
└── project3
    └── project3_name
        ├── Cargo.lock
        ├── Cargo.toml
        ├── src/
        └── target/
```


If I open the "rust/" folder via the file->open tabs in Zed and navigate to "project2/src/main.rs" autocomplete doesn't work. However, if I open the "project2/" folder via the file ->open tabs and navigate to "src/main.rs" autocomplete works. This is with or without copilot enabled.

**Comment by babylard** [comment]
Turns out I'm just dumb and didn't open my project folder inside the cargo created project. Pretty sure this is why people are having the same issue.

@siygle Let us know if this works for you

**Comment by Aspasht** [comment]
> > +1 zed 0.125.3 macos 14.3 m1, it's ok in python and elixir, but doesn't work in rust.
> 
> Yeah i'm having the same issue, all the GitHub issues I've check don't seem to help either.
> 
> edit: I noticed that I am getting the following error upon opening a rust project: ![Screenshot_20240801_110624](https://private-user-images.githubusercontent.com/75695872/354298118-bbe63ac1-7cf5-4c9d-ad87-d43d9e38769a.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzIxOTU1NzksIm5iZiI6MTczMjE5NTI3OSwicGF0aCI6Ii83NTY5NTg3Mi8zNTQyOTgxMTgtYmJlNjNhYzEtN2NmNS00YzlkLWFkODctZDQzZDllMzg3NjlhLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDExMjElMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQxMTIxVDEzMjExOVomWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTE2YWEzMGYzZjUyYzQ4NzBjNmUxMjE1NjE4YjgyYzk4MzdiNTI3N2Q3ZDdkNjgzMTBmNGU0MmMxZGFiYzMzNWImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0In0.RwYO2x2Blhrl_CnXDITIFN5JZDkITidGSOJwY0_fqDs)
> 
> Not sure if this has anything to do with it. I'm running Kubuntu if that's relevant.

getting same error 
For me, it works only when i've cargo.toml, if i i've single file ```test.rs``` autocompletion does not work.

**Comment by rajrupdasofficial** [comment]
I am facing this exact same issue . I am currently on arch linux and first I thought this is some kind of Arch issue uninstall reinstall many time but did not fix anything .
This is a bug now what I am understanding. Please fix this issue ASAP.
 
But this doesnot stopping me using zed as my editor because zed itself build with my favorite programming language Rust

**Comment by babylard** [comment]
@rajrupdasofficial Check my previous replies on this thread.

**Comment by rajrupdasofficial** [comment]
@babylard  Your solution is not working for me , I always open my projects inside the project directory where cargo.toml file located at the root . and I like to create a new project using cargo not single file based . Also after disable github copilot and uninstall reinstall rust analyzer it's still not working. 

Anyone know how to fix the issue?

**Comment by babylard** [comment]
Strange

**Comment by rajrupdasofficial** [comment]
@babylard yes tried to do everything but nothing work, not having completion is some what problematic when working with larger files

**Comment by axoking** [comment]
+1
Zed 0.170.2 on Arch Linux
installed Cargo via pacman

rust-analyzer works, it can e.g. show the types of my variables and display errors/warnings, but it can't suggest anything related to the standard library. For instance, when typing "pri", print or println don't show up. Also "use std::" gives no suggestions, however it does not throw an error when importing something from there.

**Comment by axoking** [comment]
fixed: `pacman -S rust-src` solved my issue.
